### PR TITLE
consensus/pbft: replace proposal with request context

### PR DESCRIPTION
**Pull request by bailantaotao**

### CODE REVIEW

File: consensus/pbft/types.go [hunk 1]
```diff
@@ -42,6 +42,9 @@ type RequestContexter interface {
 	// Hash retrieves hash of request.
 	Hash() common.Hash
 
+	// Hash retrieves parent hash of request.
+	ParentHash() common.Hash
```
Comment (bailantaotao) [inline consensus/pbft/types.go]: remove

File: consensus/pbft/core/final_committed.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ func (c *core) handleFinalCommitted(ev pbft.FinalCommittedEvent, p pbft.Validato
 	logger := c.logger.New("state", c.state)
 	// this block is from consensus
 	if c.subject != nil &&
-		bytes.Compare(ev.BlockHash.Bytes(), c.subject.Digest) == 0 &&
+		bytes.Compare(ev.BlockHash.Bytes(), c.subject.Digest.Bytes()) == 0 &&
```
Comment (markya0616) [inline consensus/pbft/core/final_committed.go]: ev.BlockHash == c.subject.Digest

File: consensus/pbft/events.go [hunk 3]
```diff
@@ -30,7 +30,7 @@ type ConsensusDataEvent struct {
 }
 
 type RequestEvent struct {
-	BlockContext RequestContexter
+	BlockContext Proposal
```
Comment (markya0616) [inline consensus/pbft/events.go]: Proposal Proposal?

File: consensus/pbft/core/final_committed.go (hunk 2 above)
Comment (bailantaotao) [inline consensus/pbft/core/final_committed.go]: ok

Comment (bailantaotao) [inline consensus/pbft/core/final_committed.go]: fixed in 6c7800e

File: consensus/pbft/events.go (hunk 3 above)
Comment (bailantaotao) [inline consensus/pbft/events.go]: fixed in 	c8e651c

**Comment by tailingchen** [comment]
the commit message of `df47e76` should be `common.Hash`

File: consensus/pbft/core/types_test.go [hunk 4]
```diff
@@ -70,21 +58,17 @@ func testPreprepare(t *testing.T) {
 		t.Error(err)
 	}
 
-	// if block is encoded/decoded by rlp, we cannot to compare interface data type using reflect.DeepEqual. (like BlockContext)
+	// if block is encoded/decoded by rlp, we cannot to compare interface data type using reflect.DeepEqual. (like pbft.Proposal)
 	// so individual comparison here.
-	if !reflect.DeepEqual(pp.Proposal.Header, decodedPP.Proposal.Header) {
+	if !reflect.DeepEqual(pp.Proposal.Hash(), decodedPP.Proposal.Hash()) {
```
Comment (markya0616) [inline consensus/pbft/core/types_test.go]: pp.Proposal.Hash() == decodedPP.Proposal.Hash()

Comment (bailantaotao) [inline consensus/pbft/core/types_test.go]: it will fix in next ticket

**Pull request merged**